crates/isograph_schema/src/validate_schema.rs

```diff
@@ -968,13 +980,68 @@ pub fn categorize_field_loadability<'a>(
     }
 }
 
-pub fn get_missing_arguments_and_validate_argument_types<'a>(
+lazy_static! {
+    static ref ID: FieldArgumentName = "id".intern().into();
+}
+
+fn validate_no_extraneous_arguments(
+    argument_definitions: &[&ValidatedVariableDefinition],
+    arguments: &[WithLocation<SelectionFieldArgument>],
+    location: Location,
+) -> ValidateSelectionsResult<()> {
+    let extra_arguments: Vec<_> = arguments
+        .iter()
+        .filter_map(|arg| {
+            // TODO remove this
+            // With @exposeField on Query, id field is needed because the generated
+            // query is like node(id: $id) { ... everything else }, but that
+            // id field is added in somewhere else
+
+            if arg.item.name.item.lookup() == ID.lookup() {
```

Comment:
no need to call `.lookup` here, you can compare the items directly. (If you can't, then change the type of the `FieldArgumentName` to whatever the left hand side is)